### PR TITLE
Providing alternative lookup for intent

### PR DESCRIPTION
**Pull request by sinhanurag**
This PR adds an alternative lookup for the initiation intent passed by merchants that have integrated with the Orders.V2 APIs for an AS2 transaction.

### CODE REVIEW

**Review by bluepnume — APPROVED** [review]

File: src/button/validation.js [hunk 1]
```diff
@@ -155,16 +155,18 @@ const VALIDATE_INTENTS = [
 
 export function validateOrder(orderID : string, { env, clientID, merchantID, currency, intent, vault } : OrderValidateOptions) : ZalgoPromise<void> {
     const logger = getLogger();
-    
+
     // eslint-disable-next-line complexity
     return getSupplementalOrderInfo(orderID).then(order => {
         const cart = order.checkoutSession.cart;
         const cartIntent = (cart.intent.toLowerCase() === 'sale') ? INTENT.CAPTURE : cart.intent.toLowerCase();
+        const initiationIntent = cart.initiationIntent;
```
Comment (bluepnume) [inline src/button/validation.js:163]: Will need to change the GQL query to fetch this

File: src/api/order.js [hunk 2]
```diff
@@ -423,6 +423,9 @@ type SupplementalOrderInfo = {|
                     currencyValue : string,
                     currencyCode : string
                 |}
+            |},
+            supplementary? : {|
```
Comment (bluepnume) [inline src/api/order.js:427]: One additional thing -- you'll need to add this to the actual GQL query :)

**Comment by sinhanurag** [comment]
retest

**Comment by mstuart** [comment]
Saw there was a test failure on:

image

Trying to re-run it. Maybe it's a flaky test 🤔

**Comment by mstuart** [comment]
Sweeeeet. It passed. @gregjopa @bluepnume this look okay? good to merge?

**Comment by mstuart** [comment]
Nevermind. Still need to release some back-end changes :D

**Comment by mstuart** [comment]
@bluepnume These backend changes have been released. Cool if we merge?

**Pull request merged**